text/0114-dynamic-component-configuration.md

```diff
@@ -0,0 +1,300 @@
+	@api handlers;
+
+	get configurations() {
+		return {
+			attributes: this.attributes,
+			properties: this.properties,
+			eventHandlers: this.handlers,
+		}
+	}
+}
+```
+## Working Example
+
+This example shows how we render dynamic analytics dashboard with one type of chart. The component has dashboard data that it will assign to the chart item and it renders to the screen.
+
+* *Properties*: Different chart type often requires different input property shapes. We set chart item's properties from dashboard data.
+
+* *Attributes*: For application purposes, we assign different css class values based on chart type.
+
+* *Events*: Different chart types support different operations on them. If we detect that a chart supports dashboard-wide filters, we bind the dashboard's handlers to it.
+
+```html
+<template>
+	<c-lazy-cmp lwc:bind={chartItem} lwc:dynamic={ctor}></c-lazy-cmp>
+</template>
+```
+```js
+import { LightningElement, api } from "lwc";
+
+export default class LazyCmp extends LightningElement {
+	@api dashboardId;
+	@api dashboardData;
+
+	get chartItem() {
+		return this.chartItem;
+	}
+
+	@wire(myAdapter, { id: 'someDashboardId' }) {
+		chartDefinitions({ data, error }) {
+			let chart = data.chart;
+
+			// Populate 'ctor'
+			// For simplicity assume they've been loaded async elsewhere
+			this.ctor = chart.ctor;
+			let attributes = {};
+			let props = {};
+			let eventHandlers = {};
+			
+			// Populate 'attributes', change style based on some config data
+			attributes.style = {
+				classname: chart.type === 'donut' ? 'chart-small' : 'chart-medium'
+			};
+
+			// Populate 'props', set the dimensions into charts that support them.
+			if (chart.supportsDimensions) {
+				props.dimensions = dashboardData.dimensions;
+			}
+
+			// Populate 'events', bind the event handler if it supports that
+			if (chart.canFilter) {
+				eventHandlers.onfilter = this.handleFilter;
+			}
+
+			chartItem =  {
+				attributes,
+				properties: props,
+				events: eventHandlers
+			}
+		}
+	}
+}
+```
+## Motivation
+
+Before this proposal, dynamic component creation was done using the `lwc:dynamic` directive. This directive required a component author to know component configuration details like the properties and event names at design time, which breaks down for use cases that need true dynamic component creation. For instance, in the case of property values or event handlers one would have to keep a reference to the element and set the information retrieved at runtime manually.
+
+Also, when connetedCallback called on child and the component has been inserted into DOM, but its properties/attributes/eventHandlers have not been initiated, we also need to wait for another cycle to get them rendered with the correct properties. This breaks the framework's life cycle.
+
+```js
+export default class LazyCmp extends LightningElement {
+	element;
+
+	@api moduleName;
+
+	connectedCallback() {
+		const ctor = await import(this.moduleName);
```

Comment:
invalid js

Reply:
Is this because await must be in an async function?